### PR TITLE
Improve filter bitmask generator function all in one with sse/avx2/avx512

### PR DESCRIPTION
**Pull request by jinjunzh**
Co-authored-by: @vesslanjin jun.i.jin@intel.com

Changelog category (leave one):
- Performance Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Improve filter bitmask generator function all in one with sse/avx2/avx512 instructions

Detailed description / Documentation draft:
This optimization is to add an unified filter bitmask generator function which is called in ColumnFixedString, ColumnVector, ColumnCommon, ColumnDecimal.cpp. This function is to generate  64bits mask for potential consecutive values.

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]
LGTM.

Interesting that it is using more aggressive assumption about the distribution of data - the optimization will be in effect if 64 consecutive values are either 0 or 1 (was: 16).

In fact it's expected to be beneficial.
And as soon as performance tests are green we can merge.

File: src/Columns/ColumnsCommon.h [hunk 1]
```diff
@@ -15,6 +20,40 @@ namespace ErrorCodes
...
+        | ((static_cast<UInt64>(_mm_movemask_epi8(_mm_cmpeq_epi8(
+        _mm_loadu_si128(reinterpret_cast<const __m128i *>(bytes64 + 32)), zero16))) << 32) & 0xffff00000000)
+        | ((static_cast<UInt64>(_mm_movemask_epi8(_mm_cmpeq_epi8(
+        _mm_loadu_si128(reinterpret_cast<const __m128i *>(bytes64 + 48)), zero16))) << 48) & 0xffff000000000000);
+#else
+    UInt64 res = 0;
+    const UInt8 * pos = bytes64;
+    const UInt8 * end = pos + 64;
+    for (; pos < end; ++pos)
+        res |= ((*pos == 0)<<(pos-bytes64));
```
Comment (alexey-milovidov) [inline src/Columns/ColumnsCommon.h:52]: Unfortunately, this code is wrong and it breaks all non-x86 builds.

Comment (alexey-milovidov) [reply src/Columns/ColumnsCommon.h:52]: #31574

Comment (vesslanjin) [reply src/Columns/ColumnsCommon.h:52]: Noticed. thanks for info.

**Pull request merged**